### PR TITLE
Fix: Invalid regular expression: missing / #1110

### LINKED ISSUE #1110: Invalid regular expression: missing /
Since 1 or 2 hours, all videos passed on ytdl-core return this "Invalid regular expression" error

Stack trace:
```
evalmachine.<anonymous>:11
"mfZru","cfYAzm",1897615910,414154674,"IluoVc5",null,1,"XsmT",-659785013,577478260,-419211961,1087899650,/,,[/,913,/](,)};Wka(ncode);
                                                                                                         ^

SyntaxError: Invalid regular expression: missing /
    at new Script (node:vm:100:7)
    at exports.decipherFormats (D:\Documents\GitHub\ReAllToMP3\node_modules\ytdl-core\lib\sig.js:116:51)
    at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Promise.all (index 0)
    at async exports.getInfo (D:\Documents\GitHub\ReAllToMP3\node_modules\ytdl-core\lib\info.js:401:17)
```

ytdl-core version: 4.11.0

Edit: Temporary fix linked here: https://github.com/fent/node-ytdl-core/issues/1110#issuecomment-1219856316

### LINKED ISSUE #1113: SyntaxError: Missing catch or finally after try
**Code :-**
```
const ytdl = require('ytdl-core');

ytdl.getInfo('http://www.youtube.com/watch?v=aqz-KE-bpKQ')
.then( d => console.log(d))
.catch(e => console.error(e));
```

**Expected Behaviour**
log the video info

**Ouptut on Console**
```
evalmachine.<anonymous>:24
c[72],c[6]),(0,c[85])(c[11]),(0,c[41])(c[39],c[28],(0,c[66])()),(0,c[78])(c[Math.pow(6,2)+122+-119],c[new Date("Wednesday December 31 1969 19:00:14 CDT")/1E3])}catch(d){(0<c[0]||(((0,c[80])(c[60],c[88]),c[80])(c[21],c[88]),0))&&(0,c[17])((0,c[70])(c[6]),c[78],c[88],c[12])}try{2<c[91]&&(1<c[53]?(0,c[17])((0,c[80])(c[82],c[6]),c[78],c[39],c[5]):((0,c[45])(c[49],c[88]),c[78])(c[88],c[86])),(0,c[17])((0,c[85])(c[31]),c[41],c[88],c[75],(0,c[35])())}catch(d){}};Wka(ncode);
                                                                                                                                                                                                                                                                                                                                                                                                                                                                          ^

SyntaxError: Missing catch or finally after try
    at new Script (node:vm:100:7)
    at exports.decipherFormats (/storage/emulated/0/c/ytdl/node_modules/ytdl-core/lib/sig.js:116:51)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 0)
    at async exports.getInfo (/storage/emulated/0/c/ytdl/node_modules/ytdl-core/lib/info.js:401:17)

Node.js v18.7.0
```

### PR DESCRIPTION
**Pull request by dskaiser82**
I didn't write this code fix and give the credit to @GreepTheSheep

Temporary Fix for YouTube's new signature which is linked to this issue
https://github.com/fent/node-ytdl-core/issues/1110

Please review for approval

closes #1110 
closes #1113

### CODE REVIEW

**Comment by GreepTheSheep** [comment]
ty for making this PR

**Comment by dskaiser82** [comment]
Greep while we're waiting for this PR to be merged  in I will be using the fork you made!

**Comment by richardabear** [comment]
Thanks for this guys. I'll be moving to this as well for now. hopefully this gets merged in soon.

**Review by Leomotors — APPROVED** [review]

**Comment by dskaiser82** [comment]
Looks like I have one approval but need 2 more approvals

**Review by JoshuaDarron — APPROVED** [review]

**Review by dsnsgithub — APPROVED** [review]
This pull request solves both https://github.com/fent/node-ytdl-core/issues/1110 and https://github.com/fent/node-ytdl-core/issues/1113

**Comment by dskaiser82** [comment]
We have some approvals but this is the blocker:
"First-time contributors need a maintainer to approve running workflows. [Learn more.](https://docs.github.com/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks)
1 successful check"

Who is a maintainer?

**Comment by dskaiser82** [comment]
Maybe 
TimeForANinja
[VoltrexMaster](https://github.com/VoltrexMaster)
[fent](https://github.com/fent)

Can help as maintainers?

**Comment by arcovoltaico** [comment]
It seems the project needs more maintainers or at least in different time zones as this critical issue has made the library unusable for a day. Anyway impressive effort solving it, thanks so much!

**Comment by Trott** [comment]
On Unix-like operating systems (including macOS), if you want to apply this PR to  your existing 4.11.0 installed version:

```
cd node_modules/ytdl-core
curl -L https://github.com/fent/node-ytdl-core/pull/1111.patch  | patch
```

`patch` will ask you what file you want to patch. Type `lib/sig.js`. (There's probably a CLI flag or alternative to `patch` that would cause it to correctly detect the file.)

**Comment by RyloRiz** [comment]
It's been nearly an additional twelve hours. This is a breaking change and needs to be fixed ASAP.

**Comment by Trott** [comment]
> It's been nearly an additional twelve hours. This is a breaking change and needs to be fixed ASAP.

This is an open source project that is being provided to you for free. No one asked you to use it, no one sold it to you, and most importantly, no one owes you anything. The license is clear on this: "THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED..."

If you really need  this "fixed ASAP, you can do it yourself.  The code is right there in this pull request. Here's one of many ways to hot-patch and existing installation.

```
cd node_modules/ytdl-core
curl -L https://github.com/fent/node-ytdl-core/pull/1111.patch | patch lib/sig.js
```

**Comment by Trott** [comment]
@dskaiser82 @GreepTheSheep This repository uses `semantic-release` so you'll probably need to edit the first line of the commit message to trigger a release. Maybe use `git commit --amend` to change it to `fix: adjust for YouTube's new signature` and then force push to the PR branch?

**Comment by dskaiser82** [comment]
@Trott Agree with you btw on the open source comment, and the multiple temp solutions out there.
I'll update the commit, but I dont think thats the blocker now.

It's "2 workflows awaiting approval
First-time contributors need a maintainer to approve running workflows. [Learn more.](https://docs.github.com/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks)
1 successful check"

I dont think we have any maintainers paying attention to the PR.....even so yes I can ammend the commit

**Comment by Trott** [comment]
> I'll update the commit, but I dont think thats the blocker now.

Yeah, I don't think it's a blocker at the moment. I'm just saying that you might want to have the commit message formatted the way the tool will expect it so that it doesn't end up delaying things later on.

> I dont think we have any maintainers paying attention to the PR

@fent @TimeForANinja

**Comment by dskaiser82** [comment]
"A repository owned by a personal account has two permission levels: the repository owner and collaborators. For more information, see "[Permission levels for a personal account repository](https://docs.github.com/en/articles/permission-levels-for-a-user-account-repository)."

Need that "write access".

**Comment by dskaiser82** [comment]
@Trott When the Merge button shows, we will do a squashed commit the GitHub GUI let's the user do.  We can squash then.
We have the same flow at work.


This is an example of a squashed commit:

feat: add IPv6 block rotating (#713)
    
    * IPv6 Rotating ^-^
    
    * linter got mad at missing semicolon nice
    
    * Fixing format
    
    Fixing the format of files so eslint does not throw any errors.
    
    * adding colon
    
    * added test for IPv6 Block

**Comment by mtripg6666tdr** [comment]
btw using close keyword in pr description like "close #issueid" will make a link to the original issue and automatically close the issue, it might be better if so

**Comment by 73nici** [comment]
Ty for being so fast with this fix 🚀

**Comment by Trott** [comment]
@TimeForANinja Anything else that needs to happen here before this can be merged?

**Comment by 0xC4aE1e5** [comment]
@Trott I don't think so. I used it with my app and it works very well.

**Comment by Trott** [comment]
> @Trott I don't think so. I used it with my app and it works very well.

Thanks, but I was addressing @TimeForANinja specifically because they are (AFACT) the one person who has been active in this pull request who has the permission to merge the commit. It was a hopefully-polite way of saying "bump, please--can we merge this now?"

**Comment by TimeForANinja** [comment]
> @TimeForANinja Anything else that needs to happen here before this can be merged?

just one thing @Trott and @dskaiser82
if it's possible to get this resolved by doing some minor, quick updates to the `cutAfterJSON` method i'd prefer it

**Comment by TimeForANinja** [comment]
tried some stuff
you can find my changes [here](https://github.com/fent/node-ytdl-core/compare/master...TimeForANinja:cut_json_fix?expand=1)
it worked for the hand full of requests i did - feel free to comment on it

**Comment by dskaiser82** [comment]
@TimeForANinja 

I did this PR from @GreepTheSheep 's Fork.  I've personally been using the fork and know this code update works.  With that being said, we can go with your changes instead and I can close this PR.

As you are a maintainer just advise what you'd like to do.

cc: @Trott

**Comment by gatecrasher777** [comment]
One of the array elements found in a base.js n-code extraction today is  

```javascript
    ... -1816574795, '",;/[;', function ...
```
which would break cutAfterJSON because you would have an orphan double quote.  Perhaps if single quotes and back quotes  are also added as string markers, that would do it.

**Comment by dskaiser82** [comment]
And I get it.....  @fent  makes some thing awesome that devs use... Coders like @TimeForANinja help maintain it.  Countless hours are put in to make it better and keep it a live....Noone is getting paid.....over time energy and momentum is lost....

But for the devs that do give a shit about and want to help....puts us in an frustrating situation.....which we have options....
Anyways I get it....

File: lib/sig.js
Comment (dskaiser82) [inline lib/sig.js:66]: Any changes asked for these lines?

File: lib/sig.js
Comment (dskaiser82) [inline lib/sig.js:68]: Any changes requested for these lines

File: lib/sig.js
Comment (dskaiser82) [inline lib/sig.js:67]: Any changes requested for these lines

**Comment by TimeForANinja** [comment]
replaced by #1126 
still ty for the pr and finding the problem

**Comment by wukko** [comment]
Ironically enough, this pr actually fixes the current regex issue (#1131), unlike #1126, which is already throwing errors again.

**Review by pixelblob — COMMENTED** [review]
Works well for me

**Pull request closed without merging**
